code/game/machinery/computer/camera.dm

```diff
@@ -132,7 +132,7 @@
 	var/list/visible_turfs = list()
 
 	// Is this camera located in or attached to a living thing? If so, assume the camera's loc is the living thing.
-	var/cam_location = isliving(active_camera.loc) ? active_camera.loc : active_camera
+	var/cam_location = (isliving(active_camera.loc) || ismecha(active_camera.loc)) ? active_camera.loc : active_camera
```

Comment:
What's the point of these if-checks? We only ever use cam_location inside a get_turf, and get_turf always gets the turf regardless of how many containers its in

Reply:
iirc this makes it so the camera feed isn't taken from the camera itself(objects inside another object have z = 0, so that makes a blackscreen instead of the camera feed) but the object it's in

Reply:
Yeah it grabs the turf, so why check for isliving and ismecha? If you scroll down a bit you can see it always grabs the turf, so it's pointless